src/custom/hooks/useRecentActivity.ts

```diff
@@ -259,3 +267,17 @@ export function groupActivitiesByDay(activities: ActivityDescriptors[]): Activit
     return { date: new Date(timestamp), activities: mapByTimestamp[timestamp] }
   })
 }
+
+export function useRecentActivityLastPendingOrder() {
+  const allRecentActivity = useRecentActivity()
+  const lastPendingOrder = useMemo(() => {
+    const pendings = allRecentActivity.filter(isPending)
+    const lastOrder = pendings[pendings.length - 1]
+
+    if (!lastOrder || !isAnOrder(lastOrder)) return null
+
+    return lastOrder
+  }, [allRecentActivity])
+
+  return lastPendingOrder
+}
```

Comment:
Nice, very clean 👍 

One last nitpick: You can return the `useMemo`, no need for `lastPendingOrder` parameter